usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -13838,18 +17314,23 @@ static CK_RV update_ep11_attrs_from_blob(STDLL_TokData_t *tokdata,
     if (!ep11_data->pkey_wrap_supported)
         num_ibm_attrs -= 2;
 
-    if (template_attribute_get_non_empty(tmpl, CKA_IBM_OPAQUE,
+    if (template_attribute_get_non_empty(key_obj->template, CKA_IBM_OPAQUE,
```

Comment:
I would add a check if (key_obj == NULL) at the beginning of the routine, although all callers provide a valid key_obj so far. But if it would be NULL, then referencing key_obj->template as well as calling RETRY_REENC_BLOB_START below with a null key_obj would lead to errors.

Reply:
Hm, well, this is an internal routine.  In internal routines we usually don't check all the arguments passed to us.
Calling update_ep11_attrs_from_blob() with NULL as key_obj does not make any sense, so I see this as a coding error if one does so. 
When I look at the callers, then the caller has already touched the passed key_obj pointer before calling this function, so it would have crashed already earlier.
So I would tend to lave this as is.